File: bookshelf/src/mfqbxssc/java/com.dingyue.bookshelf/BookDetailAdapter.java
package com.dingyue.bookshelf;

import android.content.Context;
import android.support.v4.view.PagerAdapter;
import android.text.TextUtils;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ImageView;
import android.widget.TextView;

import com.bumptech.glide.Glide;
import com.bumptech.glide.load.engine.DiskCacheStrategy;

import net.lzbook.kit.data.bean.Book;
import net.lzbook.kit.utils.AppLog;
import net.lzbook.kit.utils.AppUtils;
import net.lzbook.kit.utils.Tools;

import java.util.ArrayList;
import java.util.List;

/**
 * Created by Administrator on 2017\8\2 0002.
 */

public class BookDetailAdapter extends PagerAdapter {


    List<Book> lists = new ArrayList<>();
    private Context context;

    public BookDetailAdapter( Context context) {
        this.context = context;
    }

    public void setBooks(List<Book> lists){
        if(this.lists != null && this.lists.size() == 0){
            this.lists.clear();
        }
        this.lists.addAll(lists);
        notifyDataSetChanged();
    }

    @Override
    public int getCount() {
        if (lists != null) {
            return lists.size();
        }
        return 0;
    }

    @Override
    public boolean isViewFromObject(View arg0, Object arg1) {
        // TODO Auto-generated method stub
        return arg0 == arg1;
    }

    @Override
    public Object instantiateItem(ViewGroup container, int position) {
        View view = LayoutInflater.from(container.getContext())
                .inflate(R.layout.item_book_detail, container, false);
        container.addView(view);
        TextView mAuthor = (TextView) view.findViewById(R.id.book_shelf_author);
        TextView mBookName = (TextView) view.findViewById(R.id.book_shelf_name);
        ImageView mIv = (ImageView) view.findViewById(R.id.book_shelf_image);

        TextView mNewChapter = (TextView) view.findViewById(R.id.book_shelf_new);
        TextView mUpdateTime = (TextView) view.findViewById(R.id.book_shelf_update_time);

        Book book = lists.get(position);
        mAuthor.setText("作者：" + book.author);
        mBookName.setText(book.name);
        mUpdateTime.setText(Tools.compareTime(AppUtils.formatter, book
                .last_updatetime_native));

        AppLog.e("uuu", book.last_chapter_name + "===");
        mNewChapter.setText("最新章节：" + book.last_chapter_name);
        if (!TextUtils.isEmpty(book.img_url)) {
            Glide.with(context).load(book.img_url).placeholder(R.drawable.icon_book_cover_default)
                    .error((R.drawable.icon_book_cover_default)).diskCacheStrategy(DiskCacheStrategy.ALL).into(mIv);
        } else {
            Glide.with(context).load(R.drawable.icon_book_cover_default).into(mIv);
        }


        return view;
    }

    @Override
    public void destroyItem(ViewGroup container, int position, Object object) {
        container.removeView((View) object);
    }
}

File: bookshelf/src/mfqbxssc/java/com.dingyue.bookshelf/BookDetailAdapter.kt
package com.dingyue.bookshelf

import android.content.Context
import android.support.v4.view.PagerAdapter
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import com.bumptech.glide.Glide
import com.bumptech.glide.load.engine.DiskCacheStrategy
import kotlinx.android.synthetic.mfqbxssc.item_book_detail.view.*
import net.lzbook.kit.data.bean.Book
import net.lzbook.kit.utils.AppLog
import net.lzbook.kit.utils.AppUtils
import net.lzbook.kit.utils.Tools
import java.util.*

/**
 * Created by Administrator on 2017\8\2 0002
 */

class BookDetailAdapter(private val context: Context) : PagerAdapter() {

    private var books: ArrayList<Book> = ArrayList()

    fun update(books: List<Book>) {
        this.books.clear()
        this.books.addAll(books)
        notifyDataSetChanged()
    }

    override fun getCount(): Int = books.size

    override fun isViewFromObject(arg0: View, arg1: Any): Boolean {
        return arg0 === arg1
    }

    override fun instantiateItem(container: ViewGroup, position: Int): Any {
        val view = LayoutInflater.from(context)
                .inflate(R.layout.item_book_detail, container, false)
        container.addView(view)

        val book = books[position]
        val bookAuthor = "作者：${book.author}"
        view.txt_book_author.text = bookAuthor
        view.txt_book_name.text = book.name
        view.txt_book_update_time.text = Tools.compareTime(AppUtils.formatter, book
                .last_updatetime_native)

        val latestChapter = "最新章节：${book.last_chapter_name}"
        view.txt_book_chapter.text = latestChapter

        if (book.img_url.isNotEmpty()) {
            Glide.with(context)
                    .load(book.img_url)
                    .placeholder(R.drawable.icon_book_cover_default)
                    .error(R.drawable.icon_book_cover_default)
                    .diskCacheStrategy(DiskCacheStrategy.ALL)
                    .into(view.img_book_cover)
        } else {
            Glide.with(context)
                    .load(R.drawable.icon_book_cover_default)
                    .into(view.img_book_cover)
        }

        return view
    }


    override fun destroyItem(container: ViewGroup, position: Int, `object`: Any) {
        container.removeView(`object` as View)
    }

    override fun getItemPosition(`object`: Any?): Int {
        return PagerAdapter.POSITION_NONE
    }
}